walkthroughs/week-5-tdd/intro/src/main/java/com/google/sps/Greeter.java

```diff
@@ -22,6 +22,7 @@
    * Returns a greeting for the given name.
    */
   public String greet(String name) {
-    return "Hello " + name;
+    String newName = name.trim().replaceAll("[ @#$% ]","");
```

Comment:
I'm not able to test it, but I believe that this will remove spaces from the name because there are space characters inside the brackets.  Is that what you want?  If so, please remove one of the redundant spaces from the regex.

Reply:
Also, please go ahead and add a test case for names with internal spaces to ensure that this does what you want!

Reply:
Good eye; the space is intentional because one of the given test cases is to check if  whitespace is removed.

Reply:
Yes, but the line as you have it now will convert "John Doe" into "JohnDoe", which I don't think is desirable.  And even if that *is* what you want, you should use `"[@#$% ]"` with only one space, rather than `"[ @#$% ]"` with two.

Reply:
I will change that right now and add a test case.  Thank you.